packages/manager/src/hooks/useDetermineUnreachableIPs.ts

```diff
@@ -0,0 +1,182 @@
+    vpcInterfaces?.[0];
+
+  const { data: vpcLinodeIsAssignedTo } = useVPCQuery(
+    linodeInterfaceWithVPC?.vpc?.vpc_id ?? -1,
+    Boolean(vpcInterfaces?.length) && enabled
+  );
+
+  // For Linode Interfaces, a VPC only Linode is a VPC interface that is the default route for ipv4
+  // but doesn't have a nat_1_1 val
+  const isVPCOnlyLinodeInterface = Boolean(
+    linodeInterfaceWithVPC?.default_route.ipv4 &&
+      !linodeInterfaceWithVPC?.vpc?.ipv4?.addresses.some(
+        (address) => address.nat_1_1_address
+      )
+  );
+
+  const isUnreachablePublicIPv4LinodeInterface =
+    isVPCOnlyLinodeInterface ||
+    !interfaces?.interfaces.some((iface) => iface.default_route.ipv4);
+
+  // public IPv6 is (currently) not reachable if Linode has no public interfaces
+  const isUnreachablePublicIPv6LinodeInterface = !interfaces?.interfaces.some(
+    (iface) => iface.public
+  );
+
+  return {
+    isUnreachablePublicIPv4LinodeInterface,
+    isUnreachablePublicIPv6LinodeInterface,
+    linodeInterfaceWithVPC,
+    vpcLinodeIsAssignedTo,
+  };
+};
+
+/**
+ * Legacy Config Interface equivalent to useDetermineUnreachableIPsLinodeInterface
+ * Returns whether the public IPv4/IPv6 are reachable, the VPC config interface, and the
+ * VPC associated with that interface
+ */
+export const useDetermineUnreachableIPsConfigInterface = (
+  linodeId: number,
+  enabled: boolean = true
+) => {
+  const { data: configs } = useAllLinodeConfigsQuery(linodeId, enabled);
+  let interfaceWithVPC: Interface | undefined;
+
+  const configWithVPCInterface = configs?.find((config) => {
+    const interfaces = config.interfaces;
+
+    const _interfaceWithVPC = interfaces?.find(
+      (_interface) => _interface.purpose === 'vpc'
+    );
+
+    if (_interfaceWithVPC) {
+      interfaceWithVPC = _interfaceWithVPC;
+    }
+
+    return config;
+  });
+
+  const primaryInterfaceIndex = getPrimaryInterfaceIndex(
+    configWithVPCInterface?.interfaces ?? []
+  );
+
+  const vpcInterfaceIndex = configWithVPCInterface?.interfaces?.findIndex(
+    (_interface) => _interface.id === interfaceWithVPC?.id
+  );
+
+  const { data: vpcLinodeIsAssignedTo } = useVPCQuery(
+    interfaceWithVPC?.vpc_id ?? -1,
+    Boolean(interfaceWithVPC) && enabled
+  );
+
+  // A VPC-only Linode is a Linode that has at least one primary VPC interface (either explicit or implicit) and purpose vpc and no ipv4.nat_1_1 value
+  const isVPCOnlyLinode = Boolean(
+    (interfaceWithVPC?.primary ||
+      primaryInterfaceIndex === vpcInterfaceIndex) &&
+      !interfaceWithVPC?.ipv4?.nat_1_1
+  );
+
+  const hasConfigInterfaces =
+    configWithVPCInterface?.interfaces &&
+    configWithVPCInterface?.interfaces.length > 0;
+
+  // For legacy config interfaces, if a Linode has no interfaces, the API automatically provides public connectivity.
+  // IPv6 is unreachable if the Linode has interfaces, but none of these interfaces is a public interface
```

Comment:
```suggestion
  // IPv6 is unreachable if the Linode has interfaces and none of these interfaces are a public interface
```
Nit - typo + does "and" make more sense here?